xbmc/guilib/listproviders/DirectoryProvider.cpp

```diff
@@ -273,11 +273,21 @@ bool CDirectoryProvider::Update(bool forceRefresh)
   {
     CLog::Log(LOGDEBUG, "CDirectoryProvider[{}]: refreshing..", m_currentUrl);
     if (m_jobID)
-      CServiceBroker::GetJobManager()->CancelJob(m_jobID);
+    {
+      if (!m_updatePending)
```

Comment:
This is not atomic right?
I mean...there's still a chance you can have two concurrent jobs running. Wouldn't it be better to introduce a lock?

Reply:
Maybe try lock otherwise bail

Reply:
There is already a lock for the provider. `m_updatePending` will only be read/written while that mutex is locked.